From: Daniel Baker <dbaker@mozilla.com>
Date: Mon, 26 Aug 2024 15:13:25 -0600
Subject: (tmp-cherry-pick) Revert "Temporary add
 'RTPVideoHeaderH264::nalus_length'." (a0b22af9e1)

This reverts commit 04dd95fcac549fbdc330cee1de65074961db5934.

Reason for revert: code has been updated

Original change's description:
> Temporary add 'RTPVideoHeaderH264::nalus_length'.
>
> This is a forward fix for https://webrtc-review.googlesource.com/c/src/+/354622 that breaks client code using nalus_length.
>
> No-Try: true
> Change-Id: Ic0fc41696e408adefe4eb8792150a64b1eab49da
> Bug: None
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/354840
> Reviewed-by: Danil Chapovalov <danilchap@webrtc.org>
> Commit-Queue: Jeremy Leconte <jleconte@google.com>
> Owners-Override: Jeremy Leconte <jleconte@google.com>
> Cr-Commit-Position: refs/heads/main@{#42493}

Bug: None
Change-Id: I1b65fe94ca07efdb8c7643e2ac46517050095018
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/354860
Commit-Queue: Jeremy Leconte <jleconte@google.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Reviewed-by: Philip Eliasson <philipel@webrtc.org>
Reviewed-by: Danil Chapovalov <danilchap@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#42496}
---
 modules/rtp_rtcp/source/video_rtp_depacketizer_h264.cc  | 2 --
 modules/video_coding/codecs/h264/include/h264_globals.h | 2 --
 2 files changed, 4 deletions(-)

diff --git a/modules/rtp_rtcp/source/video_rtp_depacketizer_h264.cc b/modules/rtp_rtcp/source/video_rtp_depacketizer_h264.cc
index 07a9e74d65..e6b206cdd7 100644
--- a/modules/rtp_rtcp/source/video_rtp_depacketizer_h264.cc
+++ b/modules/rtp_rtcp/source/video_rtp_depacketizer_h264.cc
@@ -221,7 +221,6 @@ absl::optional<VideoRtpDepacketizer::ParsedRtpPayload> ProcessStapAOrSingleNalu(
     }
 
     h264_header.nalus.push_back(nalu);
-    ++h264_header.nalus_length;
   }
 
   return parsed_payload;
@@ -280,7 +279,6 @@ absl::optional<VideoRtpDepacketizer::ParsedRtpPayload> ParseFuaNalu(
   h264_header.nalu_type = original_nal_type;
   if (first_fragment) {
     h264_header.nalus = {nalu};
-    h264_header.nalus_length = 1;
   }
   return parsed_payload;
 }
diff --git a/modules/video_coding/codecs/h264/include/h264_globals.h b/modules/video_coding/codecs/h264/include/h264_globals.h
index 64a382935e..dac36dd734 100644
--- a/modules/video_coding/codecs/h264/include/h264_globals.h
+++ b/modules/video_coding/codecs/h264/include/h264_globals.h
@@ -83,8 +83,6 @@ struct RTPVideoHeaderH264 {
   // The packetization type of this buffer - single, aggregated or fragmented.
   H264PacketizationTypes packetization_type;
   std::vector<NaluInfo> nalus;
-  // TODO - jleconte: Remove 'nalus_length' once all the code has been migrated.
-  size_t nalus_length;
   // The packetization mode of this transport. Packetization mode
   // determines which packetization types are allowed when packetizing.
   H264PacketizationMode packetization_mode;
